test/extended/networking/commatrix.go

```diff
@@ -0,0 +1,218 @@
+	// Will be excluded in diff generatation between documented and generated comMatrix.
+	MNOExcludedMasterComDetails = []types.ComDetails{
+		{
+			Direction: "Ingress",
+			Protocol:  "TCP",
+			Port:      80,
+			NodeRole:  "master",
+			Service:   "router-default",
+			Namespace: "openshift-ingress",
+			Pod:       "router-default",
+			Container: "router",
+			Optional:  false,
+		}, {
+			Direction: "Ingress",
+			Protocol:  "TCP",
+			Port:      443,
+			NodeRole:  "master",
+			Service:   "router-default",
+			Namespace: "openshift-ingress",
+			Pod:       "router-default",
+			Container: "router",
+			Optional:  false,
+		},
+	}
+)
+
+var _ = g.Describe("[sig-network][Feature:commatrix][Serial]", func() {
+	g.It("should be equal to documeneted communication matrix", func() {
+		// artifactsDir := filepath.Join(exutil.ArtifactDirPath(), "commatrix")
+		artifactsDir := filepath.Join(".", "commatrix")
+
+		err := os.MkdirAll(artifactsDir, 0755)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		cs, err := client.New()
+		o.Expect(err).ToNot(o.HaveOccurred())
+
+		configClient := configv1client.NewForConfigOrDie(cs.Config)
+
+		deployment := types.MNO
+		isSNO, err := isSNOCluster(configClient)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		if isSNO {
+			deployment = types.SNO
+		}
+
+		env := types.Cloud
+		isBM, err := isBMCluster(configClient)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		if isBM {
+			env = types.Baremetal
+		}
+
+		g.By("preparing for commatrices generation")
+		epExporter, err := endpointslices.New(cs)
+		o.Expect(err).ToNot(o.HaveOccurred())
+		utilsHelpers := utils.New(cs)
+
+		g.By("generating new commatrix")
+		newComMatrixCreator, err := commatrixcreator.New(epExporter, "", "", env, deployment)
+		o.Expect(err).ToNot(o.HaveOccurred())
+		newComMatrix, err := newComMatrixCreator.CreateEndpointMatrix()
+		o.Expect(err).ToNot(o.HaveOccurred())
+		newComMatrix.WriteMatrixToFileByType(utilsHelpers, "new-commatrix", types.FormatCSV, deployment, artifactsDir)
+
+		g.By("get documented commatrix")
+		fp := filepath.Join(artifactsDir, "doc-commatrix.csv")
+		createCSVFromUrl(docCommatrixUrl, fp)
+		docComMatrixCreator, err := commatrixcreator.New(epExporter, fp, types.FormatCSV, env, deployment)
+		o.Expect(err).ToNot(o.HaveOccurred())
+		docComDetailsList, err := docComMatrixCreator.GetComDetailsListFromFile()
+		o.Expect(err).ToNot(o.HaveOccurred())
+
+		g.By("Filter documented commatrix for diff generation")
+		// if cluster is a SNO exclude MNO static entries in diff generation
+		if isSNO {
+			docComDetailsList = excludeGivenStaticEntries(docComDetailsList, &types.ComMatrix{Matrix: types.MNOStaticEntries}, nil)
+		} else {
+			// Exclude specific master entries that are documented as both worker and master,
+			// and are open only on the worker node in MNO clusters.
+			docComDetailsList = excludeGivenStaticEntries(docComDetailsList, &types.ComMatrix{Matrix: MNOExcludedMasterComDetails}, nil)
```

Comment:
This seems weird. It seems like it would be better to have separate CSV (or whatever) files for the different types, or else to have a single "merged" file with another column/field indicating which types of clusters the row applies to.